pkg/controller/cluster/cluster_controller.go

```diff
@@ -142,6 +144,28 @@ type clusterController struct {
 	hostClusterName string
 }
 
+func (c *clusterController) ValidateUpdate(old runtime.Object, new runtime.Object) error {
```

Comment:
I didn't get it, why does the webhook need to be related to the controller, it should be independent.

Reply:
It references existing webhook code
https://github.com/kubesphere/kubesphere/blob/master/pkg/controller/network/ippool/ippool_controller.go

Reply:
Thanks for your patient review, I have simplified the code